src/App.tsx

```diff
@@ -15,35 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Default = '',
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+}
+
+interface FilterParams {
+  sortBy: SortBy;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortBy, isReversed }: FilterParams,
+): string[] {
+  const currentVisibleGoods = [...goods];
+
+  if (sortBy) {
+    currentVisibleGoods.sort((good1: string, good2: string) => {
+      if (sortBy === SortBy.Alphabetically) {
+        return good1.localeCompare(good2);
+      }
+
+      if (sortBy === SortBy.Length) {
+        return good1.length - good2.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (isReversed) {
+    currentVisibleGoods.reverse();
+  }
+
+  return currentVisibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortByState, setSortByState] = useState(SortBy.Default);
+  const [reversedState, setReversedState] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortBy: sortByState,
+    isReversed: reversedState,
+  });
+
+  const resetHandler = (): void => {
+    setSortByState(SortBy.Default);
+    setReversedState(false);
+  };
+
+  const isGoodsSorted = sortByState || reversedState;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortByState !== SortBy.Alphabetically,
+          })}
+          onClick={() => setSortByState(SortBy.Alphabetically)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortByState !== SortBy.Length,
+          })}
+          onClick={() => setSortByState(SortBy.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !reversedState,
+          })}
+          onClick={
+            !reversedState
+              ? () => setReversedState(true)
+              : () => setReversedState(false)
+          }
```

Comment:
you can make the code shorter 
onClick={() => setReversedState(!reversedState)}

Reply:
> you can make the code shorter onClick={() => setReversedState(!reversedState)}

When you need to set state depends on the previous one you should use functional `setState`. `setReversedState(prev => !prev)`